pkg/storage/utils/decomposedfs/spacepermissions.go

```diff
@@ -0,0 +1,129 @@
+package decomposedfs
+
+import (
+	"context"
+
+	cs3permissions "github.com/cs3org/go-cs3apis/cs3/permissions/v1beta1"
+	v1beta11 "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	ctxpkg "github.com/cs3org/reva/v2/pkg/ctx"
+	"github.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/node"
+	"google.golang.org/grpc"
+)
+
+// PermissionsChecker defines an interface for checking permissions on a Node
+type PermissionsChecker interface {
+	AssemblePermissions(ctx context.Context, n *node.Node) (ap provider.ResourcePermissions, err error)
+}
+
+// CS3PermissionsClient defines an interface for checking permissions against the CS3 permissions service
+type CS3PermissionsClient interface {
+	CheckPermission(ctx context.Context, in *cs3permissions.CheckPermissionRequest, opts ...grpc.CallOption) (*cs3permissions.CheckPermissionResponse, error)
+}
+
+// Permissions manages permissions
+type Permissions struct {
+	item  PermissionsChecker   // handles item permissions
+	space CS3PermissionsClient // handlers space permissions
+}
+
+// NewPermissions returns a new Permissions instance
+func NewPermissions(item PermissionsChecker, space CS3PermissionsClient) Permissions {
+	return Permissions{item: item, space: space}
+}
+
+// AssemblePermissions is used to assemble file permissions
+func (p Permissions) AssemblePermissions(ctx context.Context, n *node.Node) (provider.ResourcePermissions, error) {
+	return p.item.AssemblePermissions(ctx, n)
+}
+
+// CreateSpace returns true when the user is allowed to create the space
+func (p Permissions) CreateSpace(ctx context.Context, spaceid string) bool {
+	return p.checkPermission(ctx, "create-space", spaceRef(spaceid))
+}
+
+// SetSpaceQuota returns true when the user is allowed to change the spaces quota
+func (p Permissions) SetSpaceQuota(ctx context.Context, spaceid string) bool {
+	return p.checkPermission(ctx, "set-space-quota", spaceRef(spaceid))
```

Comment:
why do we need to differentiate between set quota and manage space properties?
```suggestion
	return p.checkPermission(ctx, "space.writequota", spaceRef(spaceid))
```

Reply:
That is a question I cannot answer. It has been a requirement for the ticket.